lib/hammer_cli/output/adapter/abstract.rb

```diff
@@ -69,6 +70,13 @@ def output_stream
       $stdout
     end
 
+    def displayable_fields(fields, record)
```

Comment:
Nice, thank you for extracting it to a method.

I know it's not your code, but since you're touching the area... I noticed that there's some redundancy in the **base adapter**:
https://github.com/theforeman/hammer-cli/blob/master/lib/hammer_cli/output/adapter/base.rb#L31-L34
https://github.com/theforeman/hammer-cli/blob/master/lib/hammer_cli/output/adapter/base.rb#L47
and in the **tree adapter**:
https://github.com/theforeman/hammer-cli/blob/master/lib/hammer_cli/output/adapter/tree_structure.rb#L23-L26
https://github.com/theforeman/hammer-cli/blob/master/lib/hammer_cli/output/adapter/tree_structure.rb#L34

Both adapters first filter out fields that aren't displayable and checks for the same thing later again. I think that we could either use your new method to filter them out once for good in `filter_fields` or remove that `reject` block from filter_fields. The latter is more effective because we wouldn't call `data_for_field` twice.
What do you think?

Reply:
> And let the field_filter decide which fields will be kept for rendering.

Ack, I don't care that much about which of the ways you choose.

> Also I'm not sure now if displayable_fields method should be in adapter context instead of fields filter.

I was thinking about the same. In future, we'll most likely need to detect what fields are requested to be displayed prior to fetching data from API to conditionally load more/less data (eg. when GrephQL is used). On the other hand the "data for field" aren't known prior to the API request so we'll have to split the logic.
Anyway, I think that we should do this refactoring in a separate PR. This patch is nice and simple improvement that doesn't block us from doing it later. So, let's go step by step and use your `displayable_fields` for now.